### PR TITLE
Making set algorithms conform to C++20

### PR DESCRIPTION
**Pull request by hkaiser**
This PR make sure the following algorithms conform to C++20

- [x] `set_intersection`
- [x] `set_difference`
- [x] `set_symmetric_difference`
- [x] `set_union`
- [x] `includes`
- [x] `merge`
- [x] `inplace_merge`

### CODE REVIEW

File: docs/sphinx/api/public_api.rst [hunk 1]
```diff
@@ -83,10 +83,10 @@ Functions
...
 - :cpp:func:`hpx::parallel::v1::search`
 - :cpp:func:`hpx::parallel::v1::search_n`
-- :cpp:func:`hpx::parallel::v1::set_difference`
-- :cpp:func:`hpx::parallel::v1::set_intersection`
-- :cpp:func:`hpx::parallel::v1::set_symmetric_difference`
-- :cpp:func:`hpx::parallel::v1::set_union`
+- :cpp:func:`hpx::set_difference`
+- :cpp:func:`hpx::set_intersection`
+- :cpp:func:`hpx::set_symmetric_difference`
+- :cpp:func:`hpx::set_union`
```
Comment (msimberg) [inline docs/sphinx/api/public_api.rst:89]: These can be added to the `hpx::ranges` list below as well.

**Review by msimberg — CHANGES_REQUESTED** [review]

**Comment by msimberg** [comment]
This looks like it might be related to the changes here: https://cdash.cscs.ch/testDetails.php?test=28421628&build=127919 (the same test fails with some other configurations as well)?

**Comment by hkaiser** [comment]
> This looks like it might be related to the changes here: https://cdash.cscs.ch/testDetails.php?test=28421628&build=127919 (the same test fails with some other configurations as well)?

Yep, looks like I managed to break things. I'll investigate.

File: docs/sphinx/api/public_api.rst (hunk 1 above)
Comment (hkaiser) [reply docs/sphinx/api/public_api.rst:89]: Good point. Probably :cpp:func:`hpx::incudes` as well.

**Comment by hkaiser** [comment]
> > This looks like it might be related to the changes here: https://cdash.cscs.ch/testDetails.php?test=28421628&build=127919 (the same test fails with some other configurations as well)?
> 
> Yep, looks like I managed to break things. I'll investigate.

@msimberg this should be fine now. I also managed to find a bug in `hpx::includes` that was there from the beginning \o/

**Comment by hkaiser** [comment]
Please note that I will add another commit here that will make `merge` and `inplace_merge` conforming to c++20.

**Comment by hkaiser** [comment]
@msimberg this should be fine now. Sorry for the large number of seemingly unrelated changes, apparently I did miss a couple of places when moving `is_execution_policy` to namespace hpx...

File: libs/parallelism/algorithms/tests/unit/container_algorithms/inplace_merge_range.cpp [hunk 2]
```diff
@@ -177,16 +238,162 @@ void test_inplace_merge()
...
+
+    // Test projection.
+    {
+        typedef test::test_iterator<base_iterator, IteratorTag> iterator;
+
+        sol = res = org;
+
+        DataType val;
+        hpx::ranges::inplace_merge(
+            policy, iterator(res_first), iterator(res_middle),
```
Comment (msimberg) [inline libs/parallelism/algorithms/tests/unit/container_algorithms/inplace_merge_range.cpp:299]:
```suggestion
            iterator(res_first), iterator(res_middle),
```
https://cdash.cscs.ch/viewBuildError.php?buildid=128319

**Comment by msimberg** [comment]
> @msimberg this should be fine now. Sorry for the large number of seemingly unrelated changes, apparently I did miss a couple of places when moving `is_execution_policy` to namespace hpx...

No problem at all, they're not so easy to catch without the warnings. Thanks for doing this!

**Comment by hkaiser** [comment]
@msimberg I think this is good to go now.

**Pull request merged**